mender-auth/ipc/server_test.cpp

```diff
@@ -45,10 +54,31 @@ using TestEventLoop = mender::common::testing::TestEventLoop;
 
 class ListenClientTests : public testing::Test {
 protected:
-	mtesting::TemporaryDirectory tmpdir {};
-	string test_device_identity_script {path::Join(tmpdir.Path(), "mender-device-identity")};
+	// Have to use static setup/teardown/data because libdbus doesn't seem to
+	// respect changing value of DBUS_SYSTEM_BUS_ADDRESS environment variable
+	// and keeps connecting to the first address specified.
+	static void SetUpTestSuite() {
+		string dbus_sock_path = "unix:path=" + tmp_dir_.Path() + "/dbus.sock";
+		dbus_daemon_proc_.reset(
+			new procs::Process {{"dbus-daemon", "--session", "--address", dbus_sock_path}});
+		dbus_daemon_proc_->Start();
+		// give the DBus daemon time to start and initialize
+		std::this_thread::sleep_for(chrono::seconds {1});
+
+		setenv("DBUS_SYSTEM_BUS_ADDRESS", dbus_sock_path.c_str(), 1);
+	};
+
+	static void TearDownTestSuite() {
+		dbus_daemon_proc_->EnsureTerminated();
+		unsetenv("DBUS_SYSTEM_BUS_ADDRESS");
+	};
 
 	void SetUp() override {
+#if defined(__has_feature)
```

Comment:
Omg, I didn't know one could do this! This is much better than what I'm doing here: https://github.com/mendersoftware/mender/pull/1394/commits/cf354f7b6afeb330311e320ba4f34a1898de5544